### PR TITLE
buildtool integration

### PR DESCRIPTION
**Pull request by davidbuzinski**

### CODE REVIEW

**Comment by davidbuzinski** [comment]
@mw-hrastega It looks like the docs that show up on the marketplace are in the overview.md file. I didn't update that here, but hopefully we can leverage what we have from the GitHub Actions Readme.

File: tasks/run-matlab-build/v0/task.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+    "": "https://raw.githubusercontent.com/Microsoft/azure-pipelines-task-lib/master/tasks.schema.json",
+    "id": "c61c6fcc-00f1-4aa0-8513-31aad3340512",
+    "name": "RunMATLABBuild",
+    "friendlyName": "Run MATLAB Build",
+    "description": "Run builds using the MATLAB build tool. Use this command to run the tasks in the plan returned by a file named buildfile.m in the root of your repository. To use the run-build command, you need MATLAB R2022b or a later release.",
```
Comment (mcafaro) [inline tasks/run-matlab-build/v0/task.json:6]: Should we use singular instead of plural here?

"Run a build using the MATLAB build tool."

**Review by mcafaro — APPROVED** [review]

File: tasks/run-matlab-build/v0/task.json (hunk 1 above)
Comment (mw-hrastega) [reply tasks/run-matlab-build/v0/task.json:6]: 1) I agree with Mark. "a build" is more accurate.
2) command  --> task (x2)
3) run the software-build tasks specified in a file named buildfile.m.

**Comment by mw-hrastega** [comment]
> @mw-hrastega It looks like the docs that show up on the [marketplace](https://marketplace.visualstudio.com/items?itemName=MathWorks.matlab-azure-devops-extension) are in the overview.md file. I didn't update that here, but hopefully we can leverage what we have from the GitHub Actions Readme.

That's right, @davidbuzinski. Are we also ready to show the new test options in overview.md? Right now, we have two open pull requests. Should I wait for them to get closed first?

**Comment by davidbuzinski** [comment]
Yeah, that seems reasonable. I forgot that the test changes need doc updates too.



> > @mw-hrastega It looks like the docs that show up on the [marketplace](https://marketplace.visualstudio.com/items?itemName=MathWorks.matlab-azure-devops-extension) are in the overview.md file. I didn't update that here, but hopefully we can leverage what we have from the GitHub Actions Readme.
> 
> That's right, @davidbuzinski. Are we also ready to show the new test options in overview.md? Right now, we have two open pull requests. Should I wait for them to get closed first?

File: tasks/run-matlab-build/v0/task.json (hunk 1 above)
Comment (davidbuzinski) [reply tasks/run-matlab-build/v0/task.json]: fixed

File: integ-test-promote.pipeline.yml [hunk 2]
```diff
@@ -285,6 +285,83 @@ stages:
...
+              f = fopen('buildlog.txt', 'a+'); fprintf(f, 'building\n'); fclose(f);
+              function testTask(~)
+
+              f = fopen('buildlog.txt', 'a+'); fprintf(f, 'testing\n'); fclose(f);
+              function deployTask(~)
+
+              f = fopen('buildlog.txt', 'a+'); fprintf(f, 'deploying\n'); fclose(f);
+              function checkTask(~)
+
+              f = fopen('buildlog.txt', 'a+'); fprintf(f, 'checking\n'); fclose(f);
```
[the remark this answers is not in the document]
Comment (davidbuzinski) [reply integ-test-promote.pipeline.yml:327]: Somehow I think this is a GitHub yaml rendering bug. Initially I was missing the `steps:` key but after that it was not running or giving feeback until I updated that vss-extension.json file.

File: tasks/run-matlab-build/v0/task.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{
+    "": "https://raw.githubusercontent.com/Microsoft/azure-pipelines-task-lib/master/tasks.schema.json",
+    "id": "c61c6fcc-00f1-4aa0-8513-31aad3340512",
+    "name": "RunMATLABBuild",
+    "friendlyName": "Run MATLAB Build",
+    "description": "Run a build using the MATLAB build tool. Use this task to run the software-build tasks specified in a file named buildfile.m in the root of your repository. To use the run-build task, you need MATLAB R2022b or a later release.",
```
Comment (mw-hrastega) [inline tasks/run-matlab-build/v0/task.json:6]: Do you like to drop "software-build"?

Comment (davidbuzinski) [reply tasks/run-matlab-build/v0/task.json:6]: I am thinking maybe just "build tasks". What's you're opinion on that?

Comment (mw-hrastega) [reply tasks/run-matlab-build/v0/task.json:6]: Fine by me. We also can say "MATLAB build tasks".

@mcafaro, what do you think?

File: integ-test-promote.pipeline.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mcafaro) [reply integ-test-promote.pipeline.yml:327]: Weird. I see it in the [raw file too](https://raw.githubusercontent.com/mathworks/matlab-azure-devops-extension/b8426320a8ddc53cd82eb8e113aac8a89b4d3b6e/integ-test-promote.pipeline.yml). If I try to highlight the lines, they act strangely like there's some invisible character or something.

File: tasks/run-matlab-build/v0/task.json (hunk 3 above)
Comment (mcafaro) [reply tasks/run-matlab-build/v0/task.json:6]: Sounds good to me.

File: tasks/run-matlab-build/v0/task.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        "Patch": 0
+    },
+    "inputs": [
+        {
+            "name": "tasks",
+            "type": "string",
+            "label": "Tasks",
+            "required": false,
+            "defaultValue": "",
+            "helpMarkDown": "Space-separated list of tasks to run. If not specified, the action runs the default tasks in the plan returned by buildfile.m as well as all the tasks on which they depend."
```
[the remark this answers is not in the document]
Comment (davidbuzinski) [reply tasks/run-matlab-build/v0/task.json]: Fixed

File: tasks/run-matlab-build/v0/task.json (hunk 3 above)
Comment (davidbuzinski) [reply tasks/run-matlab-build/v0/task.json]: updated

File: tasks/run-matlab-build/v0/task.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        }
+    ],
+    "instanceNameFormat": "Run MATLAB Build",
+    "execution": {
+        "Node10": {
+            "target": "main.js"
+        }
+    },
+    "messages": {
+        "BuildFailed": "Build failed with non-zero exit code."
```
Comment (mw-hrastega) [inline tasks/run-matlab-build/v0/task.json:32]: Could we remove the hyphen here?

https://www.mathworks.com/support/search.html?q=nonzero&fq[]=asset_type_name:documentation&fq[]=category:matlab/index&page=1

Comment (davidbuzinski) [reply tasks/run-matlab-build/v0/task.json]: fixed

**Review by mw-hrastega — APPROVED** [review]

**Pull request merged**